### ISSUE TITLE
Skip validation on automated persisted query cache hits

### ISSUE THREAD

**Issue by jakepusateri** [apollographql/apollo-server#1566]
For larger queries, validation takes significant time (50-80ms). With automatic persisted queries in 2.0, we should be able to do this step only on write to cache, and skip it if persistedQueryHit is true.

Would you be open to a PR for this?

**Comment by jakepusateri** [comment]
I implemented a proof of concept of this in our server by adding a simple in-memory lru cache of parsed query objects keyed by the md5 hash of the query. The queries were added to the cache after they passed validation. This cut cpu usage in half and reduced query times of complex queries by 40%. I'm planning on making a PR later this week.

**Comment by martijnwalraven** [comment]
@jakepusateri Those are great results! Note that we're working on a refactoring of `apollo-server-core` that will land soon: https://github.com/apollographql/apollo-server/tree/abernix/re-new-request-pipeline

So it probably make sense to wait until that has been merged to open a PR.

**Comment by abernix** [comment]
While we didn't directly link it to the APQ implementation, I did implement a parse/validated cache store in #2111 which, as you're certainly aware from your above experimentation, substantially reduced the time spent in those phases!

I'll close this since I think we've got the spirit of your request implemented, even if it wasn't to the exact letter of your request. 😄  Hope you're able to try it out!
